tests/per_provider/normal_tests/export_public_key.rs

```diff
@@ -53,16 +66,71 @@ fn import_and_export_public_key() -> Result<()> {
         111, 254, 183, 54, 229, 109, 28, 39, 22, 141, 173, 85, 26, 58, 9, 128, 27, 57, 131, 2, 3,
         1, 0, 1,
     ];
-    client.import_key(
-        key_name.clone(),
-        KeyType::RsaPublicKey,
-        Algorithm::AsymmetricSignature(AsymmetricSignature::RsaPkcs1v15Sign {
-            hash_alg: Hash::Sha256,
-        }),
-        key_data.clone(),
-    )?;
+    client.import_rsa_public_key(key_name.clone(), key_data.clone())?;
 
     assert_eq!(key_data, client.export_public_key(key_name)?);
 
     Ok(())
 }
+
+#[test]
+fn check_public_rsa_export_format() -> Result<()> {
+    let mut client = TestClient::new();
+    let key_name = String::from("check_public_rsa_export_format");
+    client.generate_rsa_sign_key(key_name.clone())?;
+    let public_key = client.export_public_key(key_name)?;
+
+    // That should not fail if the bytes are in the expected format.
+    let _public_key: RsaPublicKey = picky_asn1_der::from_bytes(&public_key).unwrap();
+    Ok(())
+}
+
+#[test]
+fn check_public_rsa_export_format2() -> Result<()> {
+    // Same test but with a RsaPublicKey
+    let mut client = TestClient::new();
+    let key_name = String::from("check_public_rsa_export_format2");
+
+    client.generate_rsa_sign_key(key_name.clone())?;
+
+    let public_key = client.export_public_key(key_name)?;
+
+    // That should not fail if the bytes are in the expected format.
+    let _public_key: RsaPublicKey = picky_asn1_der::from_bytes(&public_key).unwrap();
+    Ok(())
+}
+
+#[test]
+fn check_export_public_possible() -> Result<()> {
+    // Exporting a public key is always permitted
+    let mut client = TestClient::new();
+    let key_name = String::from("check_export_public_possible");
+
+    let key_attributes = KeyAttributes {
+        key_type: KeyType::RsaKeyPair,
+        key_bits: 1024,
+        key_policy: KeyPolicy {
+            key_usage_flags: UsageFlags {
+                sign_hash: false,
+                verify_hash: false,
+                sign_message: false,
+                verify_message: false,
+                export: false,
+                encrypt: false,
+                decrypt: false,
+                cache: false,
+                copy: false,
+                derive: false,
+            },
+            key_algorithm: Algorithm::AsymmetricSignature(AsymmetricSignature::RsaPkcs1v15Sign {
+                hash_alg: Hash::Sha256,
+            }),
+        },
+    };
+
+    client.generate_key(key_name.clone(), key_attributes)?;
+
+    let _public_key = client.export_public_key(key_name)?;
+
+    Ok(())
```

Comment:
you could set the return value of the test to `Result<Vec<u8>>` or whatever `export_public_key` returns, saves you of some needless code!

Reply:
True. I did intentionally not write good quality code as this is a test 😛